### DISCUSSION TITLE
Equivariant Multilayer Perceptrons

### DISCUSSION THREAD

**Discussion by PhilipVinc** [netket/netket#665] [Ideas]
@femtobit , @chrisrothUT  Maybe you saw already but there is this interesting article by `@ mfinzi` about implementing equivariant perceptrons in jax.
Maybe there could be ideas we can steal.

Their approach is also quite general on the groups they can represent...
They also have a type system for irreducible representations, apparently
[A Practical Method for Constructing Equivariant Multilayer Perceptrons for Arbitrary Matrix Groups](https://arxiv.org/pdf/2104.09459.pdf)

[mfinzi/equivariant-MLP ](https://github.com/mfinzi/equivariant-MLP)

**Comment by chrisrothUT** [comment]
Thanks! I actually had not seen this. It’s not the lightest read but I definitely want to understand it. On first glance it looks like a different model than the G-CNN where they somehow make the representation space more compact.
